sos/report/plugins/openvswitch.py

```diff
@@ -147,18 +166,34 @@ def setup(self):
                             "ovs-ofctl -O %s dump-ports-desc %s" % (flow, br)
                         ])
 
-        # Gather info on the DPDK mempools associated with each DPDK port
-        br_list_result = self.collect_cmd_output("ovs-vsctl -t 5 list-br")
-        if br_list_result['status'] == 0:
-            for br in br_list_result['output'].splitlines():
                 port_list_result = self.exec_cmd(
                     "ovs-vsctl -t 5 list-ports %s" % br
                 )
                 if port_list_result['status'] == 0:
                     for port in port_list_result['output'].splitlines():
-                        self.add_cmd_output(
-                            "ovs-appctl netdev-dpdk/get-mempool-info %s" % port
-                        )
+                        self.add_cmd_output([
+                            "ovs-appctl cfm/show %s" % port,
+                            "ovs-appctl qos/show %s" % port,
+                            # Not all ports are "bond"s, but all "bond"s are
+                            # a single port
+                            "ovs-appctl bond/show %s" % port,
+                        ])
+
+                        if check_dpdk:
+                            self.add_cmd_output(
+                                "ovs-appctl netdev-dpdk/get-mempool-info %s" %
+                                port
+                            )
+
+                iface_list_result = self.exec_cmd(
+                    "ovs-vsctl -t 5 list-ifaces %s" % br
+                )
+                if iface_list_result['status'] == 0:
+                    for iface in iface_list_result['output'].splitlines():
+                        if check_dpdk:
+                            self.add_cmd_output(
+                                "ovs-appctl netdev-dpdk/get-mempool-info %s" %
+                                iface)
```

Comment:
I would vote to put this whole section under the `if check_dpdk:` test. Since if `check_dpdk` is `False`, whole current section does not store anything but runs redundant commands.

Reply:
@pmoravec  I can do that